mergify_engine/squash_pull.py

```diff
@@ -0,0 +1,100 @@
+# -*- encoding: utf-8 -*-
+#
+#  Copyright © 2021 Mergify SAS
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+from mergify_engine.clients import http
+from mergify_engine import check_api
+
+
+async def squash_pull(ctxt):
+
+    ref = ctxt.pull["head"]["ref"]
+    owner = ctxt.pull["user"]["login"]
+    repo = ctxt.pull["head"]["repo"]["name"]
+
+    # get the branch
+    try:
+        branch = await ctxt.client.item(
+            f"{ctxt.base_url}/repos/{owner}/{repo}/git/ref/heads/{ref}"
+        )
+    except http.HTTPClientSideError as e:  # pragma: no cover
+        return check_api.Result(
+            check_api.Conclusion.FAILURE,
+            "Pull request can't be squashed",
+            e.message,
+        )
+
+    # get the last commit
+    commit_url = branch["object"]["url"]
```

Comment:
I would like it to actually use the message from the first commit instead and adding each commit messages in the pr into the message from the last until the last commit was added to the message (like how git normally does it).

Reply:
Hi AraHaan! Thanks for your comments. Basically, it's just a draft, and I have to improve it. When the PR is ready to review, don't hesitate to take a look at it! ;)

Reply:
> However if the pr currently does not have more than 1 commit, I would have the squash command do nothing and report that squashing is not needed because there is only a single commit in the pr.

Sorry, I meant that part by how it would work for PR's made on people's repositories that use the bot in general that it should do that when the pr's that gets that command sent in should failover basically and tell them that there is already a single commit on that pr.